apps/web/src/views/Swap/V3Swap/index.tsx

```diff
@@ -35,42 +55,40 @@ export function V3SwapForm() {
     [refresh],
   )
 
-  const finalTrade = mm.isMMBetter ? mm?.mmTradeInfo?.trade : trade
-
-  // console.debug('debug trade', {
-  //   trade,
-  //   mm,
-  //   finalTrade,
-  // })
+  const finalTrade = mmCurrentTrade.isMMBetter ? mmCurrentTrade?.mmTradeInfo?.trade : ammCurrentTrade
 
   const tradeLoaded = !isLoading
-  const price = useMemo(() => trade && SmartRouter.getExecutionPrice(trade), [trade])
+  const price = useMemo(() => ammCurrentTrade && SmartRouter.getExecutionPrice(ammCurrentTrade), [ammCurrentTrade])
 
-  const insufficientFundCurrency = useCheckInsufficientError(trade)
+  const insufficientFundCurrency = useCheckInsufficientError(ammCurrentTrade)
 
   const featureEnabled = useExperimentalFeatureEnabled(EXPERIMENTAL_FEATURES.UniversalRouter)
   const commitButton = useMemo(() => {
     if (featureEnabled) {
-      return mm?.isMMBetter ? (
-        <MMCommitButtonV2 {...mm} />
+      return mmCurrentTrade?.isMMBetter ? (
+        <MMCommitButtonV2 {...mmCurrentTrade} setLock={setLock} />
       ) : (
-        <SwapCommitButtonV2 trade={trade} tradeError={error} tradeLoading={!tradeLoaded} />
+        <SwapCommitButtonV2 trade={ammCurrentTrade} tradeError={error} tradeLoading={!tradeLoaded} setLock={setLock} />
       )
     }
```

Comment:
Shall we have a separate `useBestTrade` hook to include the compare logic? And I think we can make use of the `enabled` field from `useQuery` to pause the rfq if we're in the middle of the operation
```typescript
const [isQuotingPaused, setIsQuotingPaused] = useState(false)
const trade = useBestTrade(isQuotingPaused)

<CommitButton
  trade={trade}
  onSwap={pause}
  onFinished={unpause}
/>
``` 

Does it make sense?